src/main/groovy/com/ofg/uptodate/finder/NewVersionFinder.groovy

```diff
@@ -1,5 +1,57 @@
 package com.ofg.uptodate.finder
 
-interface NewVersionFinder {
-    List<Dependency> findNewer(List<Dependency> dependencies)
-}
\ No newline at end of file
+import com.ofg.uptodate.LoggerProxy
+import com.ofg.uptodate.UptodatePluginExtension
+import groovy.util.logging.Slf4j
+
+@Slf4j
+class NewVersionFinder {
+
+    private final LatestDependenciesProvider latestDependenciesProvider
+    private final FinderConfiguration finderConfiguration
+    private final LoggerProxy loggerProxy
+
+    private NewVersionFinder(LoggerProxy loggerProxy,
+                             RepositorySettingsProvider repositorySettingsProvider,
+                             LatestDependenciesProvider latestDependenciesProvider,
+                             UptodatePluginExtension uptodatePluginExtension) {
+
+        RepositorySettings repositorySettings = repositorySettingsProvider.getFrom(uptodatePluginExtension)
+        finderConfiguration = new FinderConfiguration(
```

Comment:
Maybe FinderConfiguration could take as params of constructor repositorySetting and uptodatePluginExtension and then build itself inside the constructor?